packages/quantic/force-app/main/default/lwc/quanticRecommendationList/__tests__/quanticRecommendationList.test.js

```diff
@@ -0,0 +1,428 @@
+        selectors.placeholder
+      );
+
+      expect(loadingPlaceholder).not.toBeNull();
+    });
+  });
+
+  describe('controller initialization', () => {
+    it('should build the buildRecommendationList controller with the proper parameters and subscribe to its state', async () => {
+      createTestComponent();
+      await flushPromises();
+
+      expect(functionMocks.buildRecommendationList).toHaveBeenCalledTimes(1);
+      expect(functionMocks.buildRecommendationList).toHaveBeenCalledWith(
+        exampleEngine,
+        {
+          options: {
+            id: defaultOptions.recommendation,
+            numberOfRecommendations: Number(
+              defaultOptions.numberOfRecommendations
+            ),
+          },
+        }
+      );
+    });
+
+    it('should build the buildResultTemplatesManager controller with the proper parameters', async () => {
+      createTestComponent();
+      await flushPromises();
+
+      expect(functionMocks.buildResultTemplatesManager).toHaveBeenCalledTimes(
+        1
+      );
+      expect(functionMocks.buildResultTemplatesManager).toHaveBeenCalledWith(
+        exampleEngine
+      );
+    });
+
+    it('should dispatch quantic__registerrecommendationtemplates event', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      const event = new CustomEvent(
+        'quantic__registerrecommendationtemplates',
+        {
+          bubbles: true,
+          detail: {
+            content: defaultOptions.recommendation,
+            conditions: [],
+          },
+        }
+      );
+      element.shadowRoot.dispatchEvent(event);
+
+      expect(functionMocks.registerTemplates).toHaveBeenCalledTimes(1);
+    });
+
+    it('should register the fields to include', async () => {
+      const expectedFormattedFieldsToInclude = ['field1', 'field2'];
+      createTestComponent();
+      await flushPromises();
+
+      expect(functionMocks.loadFieldActions).toHaveBeenCalledTimes(1);
+      expect(functionMocks.loadFieldActions).toHaveBeenCalledWith(
+        exampleEngine
+      );
+      expect(functionMocks.dispatch).toHaveBeenCalledTimes(1);
+      expect(functionMocks.dispatch).toHaveBeenCalledWith(
+        functionMocks.registerFieldsToInclude(expectedFormattedFieldsToInclude)
+      );
+    });
+
+    describe('with custom #numberOfRecommendations option', () => {
+      it('should build the buildRecommendationList controller with the custom number of recommendation value', async () => {
+        createTestComponent({
+          ...defaultOptions,
+          numberOfRecommendations: 5,
+        });
+        await flushPromises();
+
+        expect(functionMocks.buildRecommendationList).toHaveBeenCalledTimes(1);
+        expect(functionMocks.buildRecommendationList).toHaveBeenCalledWith(
+          exampleEngine,
+          {
+            options: {
+              id: defaultOptions.recommendation,
+              numberOfRecommendations: Number(5),
```

Comment:
Why do you need `Number(5)` ?

Reply:
no need, it was like this in the file so I went ham on this